src/Application/Console/Formatters/Checkstyle.php

```diff
@@ -57,7 +57,7 @@ public function format(
 
                 /** @var Details $detail */
                 foreach ($details as $detail) {
-                    $fileName = $this->getFileName($detail, $dir);
+                    $fileName = $this->getFileName($detail, $dir[0]);
```

Comment:
Hmm this ony is a bit more tricky. Don't think we can use the first dir here for it. 
As then it would shorten the filename wrongly. 

So we need to think of an approach for shortening the filename when you supply multiple paths.
This is a problem for almost all formatters

Reply:
@olivernybroe Yes you are right! This is one of the points I wanted to discuss. Actually there is three points like this which i'm concern about.

Reply:
Alright, so shortening the filename. 

So let's paint a scenario, I give php insights 4 paths, because I modified these 4 files
- `app/Console/Commands/MyTestCommand.php`
- `/Users/olivernybroe/Code/myCodeBase/app/Http/Controllers/Auth/LoginController.php`
- `routes/api.php`
- `tests/Feature/Auth/LoginControllerTest.php`

When I see the output from these files, shortening them based on the first input would result in all except the first one showing the full path (e.g `/Users/olivernybroe/Code/myCodeBase/app/Http/Controllers/Auth/LoginController.php`) which is not ideal.  
I think the most ideal shortening that we could do for this scenario would be to remove `/Users/olivernybroe/Code/myCodeBase`from all files. So basically finding the most folders which they all have in common.

Let's look at another example with files in different projects
- `/Users/olivernybroe/Code/myCodeBase/app/Http/Controllers/Auth/LoginController.php`
- `/Users/olivernybroe/Code/myOtherCodeBase/bin/phpinsights.php`

In this case the best shortening would be `/Users/olivernybroe/Code`.

So basically what we need to do is:
- Take all the paths the user has given us and get the full path for them all
- Sort the paths
- Take the first and last element of the paths
- Compare first and last element, to find longest common path

This works because we sort the paths and if we only have one path, then first and last element is just the same. When comparing the first and last element we prob. have to take into account the path separator. Because if we look at my second example, the first characters that are in common is actually `/Users/olivernybroe/Code/my`, however it should only remove `/Users/olivernybroe/Code` from the paths.


@50bhan What do you think of this approach?

Reply:
@olivernybroe It's a nice approach and i'm agree with it. Specially I don't like shortening to file name because what if we had two files with same name at different paths? So, I think this works. In case of one path, we can shortening to filename because there will be no conflict in names. In other cases we will shortening by common path which should be calculated by path separator as you mentioned.

Reply:
Sounds great. Let's go with this approach then!

Exactly, with one path, we shorten to filename, and this approach should actually result in that, as the longest common path when we only have one path, is basically the full path.